tests/zkevm/test_worst_blocks.py

```diff
@@ -0,0 +1,153 @@
+"""
+abstract: Tests zkEVMs worst-case block scenarios.
+    Tests zkEVMs worst-case block scenarios.
+
+Tests running worst-case block scenarios for zkEVMs.
+"""
+
+import pytest
+
+from ethereum_test_forks import Fork
+from ethereum_test_tools import (
+    Account,
+    Alloc,
+    Block,
+    BlockchainTestFiller,
+    Environment,
+    Transaction,
+)
+
+
+@pytest.fixture
+def iteration_count(eth_transfer_cost: int):
+    """Calculate the number of iterations based on the gas limit and intrinsic cost."""
+    return min(5, Environment().gas_limit // eth_transfer_cost)
+
+
+@pytest.fixture
+def transfer_amount():
+    """Ether to transfer in each transaction."""
+    return 1
+
+
+@pytest.fixture
+def eth_transfer_cost(fork: Fork):
+    """Transaction gas limit."""
+    intrinsic_cost = fork.transaction_intrinsic_cost_calculator()
+    return intrinsic_cost()
+
+
+def get_distinct_sender_list(pre: Alloc, iteration_count: int):
+    """Get a list of distinct sender accounts."""
+    return [pre.fund_eoa() for _ in range(iteration_count)]
+
+
+def get_distinct_receiver_list(pre: Alloc, iteration_count: int):
+    """Get a list of distinct receiver accounts."""
+    return [pre.fund_eoa(0) for _ in range(iteration_count)]
+
+
+def get_single_sender_list(pre: Alloc, iteration_count: int):
+    """Get a list of single sender accounts."""
+    sender = pre.fund_eoa()
+    return [sender for _ in range(iteration_count)]
```

Comment:
```suggestion
def get_single_sender_list(pre: Alloc):
    """Get a list of single sender accounts."""
    sender = pre.fund_eoa()
    while True:
        yield sender
```
Similar to previous suggestion.